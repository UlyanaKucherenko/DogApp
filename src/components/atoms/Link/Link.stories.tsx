import { Link } from '@mui/material';
import type { Meta, StoryObj } from '@storybook/react';

const meta = {
  title: 'Atoms/Link',
  component: Link,
  argTypes: {
    variant: {
      options: ['inherit', 'h2'],
      control: 'radio'
    },
    underline: ['none', 'hover']
  }
} satisfies Meta<typeof Link>;

export default meta;

type Story = StoryObj<typeof meta>;

export const LinkMore: Story = {
  render: () => (
    <Link href="/" variant="inherit" underline="hover">
      more
    </Link>
  )
};

export const NavLink: Story = {
  render: () => (
    <Link href="/" variant="h2" underline="none">
      Main
    </Link>
  )
};
