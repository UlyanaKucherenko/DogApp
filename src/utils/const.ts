export const menuHeader = [
  {
    title: 'Main',
    route: '/'
  },
  {
    title: 'Breeds',
    route: '/breeds'
  },
  {
    title: 'Vote',
    route: '/vote'
  },
  {
    title: 'History',
    route: '/history'
  },
  {
    title: 'Your uploads',
    route: '/your-uploads'
  }
];
