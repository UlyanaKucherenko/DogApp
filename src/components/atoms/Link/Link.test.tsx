import { composeStories } from '@storybook/react';
import * as stories from './Link.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('atoms/Link', () => {
  const { LinkMore, NavLink } = composeStories(stories);
  testSnapshot(<LinkMore />);
  testSnapshot(<NavLink />);
});
