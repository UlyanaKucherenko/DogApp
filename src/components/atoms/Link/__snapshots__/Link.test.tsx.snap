// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Link should match snapshot Component 1`] = `
.emotion-0 {
  margin: 0;
  font: inherit;
  color: #921FED;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #5F556D;
  background-color: transparent;
  font-size: 20px;
  text-transform: lowercase;
  border-radius: 6px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  white-space: nowrap;
  color: #FFCF32;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:hover {
  color: #A239F4;
}

.emotion-0:active {
  color: #A239F4;
}

<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-0"
    href="/"
  >
    more
  </a>
</div>
`;

exports[`atoms/Link should match snapshot Component 2`] = `
.emotion-0 {
  margin: 0;
  font-family: Dosis,FugazOne,Karantina,MedulaOne,Goldman,Mohave,MonomaniacOne,Raleway;
  font-weight: 300;
  font-size: 5.357142857142858rem;
  line-height: 1.2;
  color: #921FED;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #5F556D;
  background-color: transparent;
  font-size: 20px;
  text-transform: lowercase;
  border-radius: 6px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  white-space: nowrap;
  text-transform: capitalize;
  padding: 4px 10px;
}

.emotion-0:hover {
  background-color: #A239F4;
  color: #FFFFFF;
}

.emotion-0:active {
  background-color: #921FED;
  color: #FFFFFF;
  box-shadow: 3px 3px 0px 0px rgba(0, 0, 0, 1);
}

<div>
  <a
    class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone emotion-0"
    href="/"
  >
    Main
  </a>
</div>
`;
